src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -161,13 +172,11 @@ public DocumentNode ParseSchemaGraphQLDocument()
                     subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
             }
 
-            return Utf8GraphQLParser.Parse(graphqlSchema);
+            root = Utf8GraphQLParser.Parse(graphqlSchema);
```

Comment:
If `Utf8GraphQLParser.Parse(graphqlSchema)` throws an exception during parsing, then root may stay null, and the check on L62 `if (root is null)` will never be reached due to an uncaught exception. perhaps wrap ParseSchemaGraphQLDocument() in a try/catch so you can communicate any raised exceptions as DAB friendly error messages exposed through dabexception.

Reply:
wrapped.